src/sync/discovery.ts

```diff
@@ -1,62 +1,86 @@
-import { getRepositorySummary } from "./jobs";
 import { Subscription, Repositories, Repository } from "models/subscription";
 import { LoggerWithTarget } from "probot/lib/wrap-logger";
-import { sqsQueues } from "../sqs/queues";
 import { GitHubInstallationClient } from "../github/client/github-installation-client";
-import { getCloudInstallationId } from "../github/client/installation-id";
-import { DiscoveryMessagePayload } from "../sqs/discovery";
+import { GitHubAPI } from "probot";
+import { TaskPayload } from "~/src/sync/installation";
+import { DiscoveryMessagePayload } from "~/src/sqs/discovery";
+import { getCloudInstallationId } from "~/src/github/client/installation-id";
+import { sqsQueues } from "~/src/sqs/queues";
 
 /*
 * Mapping the response data into a map by repo.id as required by subscription.updateSyncState.
 */
 const mapRepositories = (repositories: Repository[]): Repositories => {
 	return repositories.reduce((obj, repo) => {
-		obj[repo.id] = { repository: getRepositorySummary(repo) };
+		obj[repo.id] = { repository: repo };
 		return obj;
 	}, {});
 };
 
-/*
-* Reset the sync count to zero.
-*/
-const resetSyncedReposCount = async (subscription: Subscription): Promise<void> => {
-	await subscription.updateSyncState({ numberOfSyncedRepos: 0 });
-};
-
 /*
 * Update the sync status of a batch of repos.
 */
-const updateSyncState = async (subscription: Subscription, repositories: Repository[]): Promise<void> => {
-	const repos = mapRepositories(repositories);
-	await subscription.updateSyncState({
-		repos
-	});
+const updateSyncState = async (subscription: Subscription, repositories: Repository[], totalNumberOfRepos: number): Promise<void> => {
+	await Promise.all([
+		subscription.updateSyncState({
+			repos: mapRepositories(repositories)
+		}),
+		subscription.update({ totalNumberOfRepos })
+	]);
 };
 
-/*
-* Continuosuly call the GitHub repo to fetch a page of repositories at a time and update there sync status until no more pages.
-*/
-const syncRepositories = async (github, subscription: Subscription, logger: LoggerWithTarget): Promise<void> => {
-	let page = 1;
-	let requestNextPage = true;
-	await resetSyncedReposCount(subscription);
-	while (requestNextPage) {
-		try {
-			const { data, hasNextPage } = await github.getRepositoriesPage(page);
-			requestNextPage = hasNextPage;
-			await updateSyncState(subscription, data.repositories);
-			logger.info(`${data.repositories.length} Repositories syncing`);
-			page++;
-		} catch (err) {
-			requestNextPage = false;
-			throw new Error(err);
-		}
+export const getRepositoryTask = async (
+	logger: LoggerWithTarget,
+	_github: GitHubAPI,
+	newGithub: GitHubInstallationClient,
+	jiraHost: string,
+	_repository: Repository,
+	cursor?: string | number,
+	perPage?: number
+): Promise<TaskPayload> => {
+	logger.debug("Repository Discovery: started");
+	const installationId = newGithub.githubInstallationId.installationId;
```

Comment:
Will there be a difference for GHE installationId or will this work for both?

Just thinking it might be handy to wrap this function instead of exposing the property.....however might pay to wait until GHE and not preempt without all the GHE details ¯\_(ツ)_/¯

Reply:
frankly, I'm not sure and didn't want to work on a solution before we know it's a problem.  I think this is fine for now.  We still have a lot of things to improve along the way.